### ISSUE TITLE
Issues with MPI

### ISSUE THREAD

**Issue by rbeucher** [coecms/cms-conda-singularity#46]
Hi @dsroberts 

I am hitting that issue while trying to launch a MPI job.

```
Loading conda/access-med-0.3
  Loading requirement: singularity
--------------------------------------------------------------------------
Open MPI tried to bind a new process, but something went wrong.  The
process was killed without launching the target application.  Your job
will now abort.

  Local host:        gadi-cpu-clx-0458
  Application name:  /g/data/xp65/public/apps/med_conda_scripts/access-med-0.3.d/bin/ilamb-run
  Error message:     hwloc_set_cpubind returned "Error" for bitmap "0,48"
  Location:          /jobfs/53639599.gadi-pbs/0/openmpi/4.1.4/source/openmpi-4.1.4/orte/mca/rtc/hwloc/rtc_hwloc.c:382
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec was unable to start the specified application as it encountered an
error:

Error code: 1
Error name: (null)
Node: gadi-cpu-clx-0458

when attempting to start process rank 0.
--------------------------------------------------------------------------
[gadi-cpu-clx-0458.gadi.nci.org.au:1505436] 9 more processes have sent help message help-orte-odls-default.txt / binding generic error
[gadi-cpu-clx-0458.gadi.nci.org.au:1505436] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages

```

@rhaegar325 would be good to see if we can fix that.

Romain

**Comment by dsroberts** [comment]
Hi @rbeucher 
Its something to do with the MPI binding.
```
Error message:     hwloc_set_cpubind returned "Error" for bitmap "0,48"
```
There are a bunch of things that can cause this, and I guess if MPI is losing track of cgroups or PBS due to singularity then that might cause this. Can you send me the PBS script? I'd like to see if I can replicate it.

**Comment by rhaegar325** [comment]
Hi, @dsroberts , Thanks for reply, our PBS script is like this:
```
#!/bin/bash

#PBS -N test
#PBS -l wd
#PBS -P iq82
#PBS -q express
#PBS -l walltime=0:20:00  
#PBS -l ncpus=14
#PBS -l mem=100GB           
#PBS -l jobfs=10GB        
#PBS -l storage=gdata/nf33+gdata/fs38+gdata/oi10+gdata/xp65+gdata/ct11
#PBS -o ./output_log
#PBS -e ./error_log

module use /g/data/xp65/public/modules
module load conda/access-med

export ILAMB_ROOT=/g/data/nf33/public/data/ILAMB/tree_builder/test
export CARTOPY_DATA_DIR=../
export BUILD_DIR=../ILAMB_RESULT/_build

rm -rf $BUILD_DIR
mpiexec -n 8 ilamb-run --config CMIP6.cfg --model_setup ./workshop_version.txt --regions global --build_dir $BUILD_DIR
```
The error above only occured when processes number is above helf of cpu numbers we apply from PBS.( in this example we apply 14 cpus and 8 processes will have that error)

Rhaegar

**Comment by dsroberts** [comment]
Hi @rhaegar325 @rbeucher 
Just letting you know I've worked this one out. Its because I"m not binding /sys/fs/cgroup into the container. I'm on the road all of tomorrow, so won't be able to fix this until Monday.

**Comment by rbeucher** [comment]
Thanks @dsroberts . No pb. We can wait.

**Comment by dsroberts** [comment]
Fixed in 1eff184
